test/integration/auth.spec.ts

```diff
@@ -321,6 +322,118 @@ describe('admin.auth', () => {
+    let expectedIat: number;
+    const expiresIn = 24 * 60 * 60 * 1000;
+    let payloadClaims: any;
+    let currentIdToken: string;
+    const uid = sessionCookieUid;
+
+    it('creates a valid Firebase session cookie', () => {
+      return admin.auth().createCustomToken(uid, {admin: true, groupId: '1234'})
+        .then((customToken) => firebase.auth().signInWithCustomToken(customToken))
+        .then((user) => user.getIdToken())
+        .then((idToken) => {
+          currentIdToken = idToken;
+          return admin.auth().verifyIdToken(idToken);
+        }).then((decodedIdTokenClaims) => {
+          expectedExp = Math.floor((new Date().getTime() + expiresIn) / 1000);
+          payloadClaims = decodedIdTokenClaims;
+          payloadClaims.iss = payloadClaims.iss.replace(
+              'securetoken.google.com', 'session.firebase.google.com');
+          delete payloadClaims.exp;
+          delete payloadClaims.iat;
+          expectedIat = Math.floor(new Date().getTime() / 1000);
+          // One day long session cookie.
+          return admin.auth().createSessionCookie(currentIdToken, {expiresIn});
+        })
+        .then((sessionCookie) => admin.auth().verifySessionCookie(sessionCookie))
+        .then((decodedIdToken) => {
+          // Check for expected expiration with +/-2 seconds of variation.
+          expect(decodedIdToken.exp).to.be.within(expectedExp - 2, expectedExp + 2);
+          expect(decodedIdToken.iat).to.be.within(expectedIat - 2, expectedIat + 2);
+          // Not supported in ID token,
+          delete decodedIdToken.nonce;
+          // exp and iat may vary depending on network connection latency.
+          delete decodedIdToken.exp;
+          delete decodedIdToken.iat;
+          expect(decodedIdToken).to.deep.equal(payloadClaims);
+        });
+    }).timeout(5000);
+
+    it('creates a revocable session cookie', () => {
+      let currentSessionCookie: string;
+      return admin.auth().createCustomToken(uid)
+        .then((customToken) => firebase.auth().signInWithCustomToken(customToken))
+        .then((user) => user.getIdToken())
+        .then((idToken) => {
+          // One day long session cookie.
+          return admin.auth().createSessionCookie(idToken, {expiresIn});
+        })
+        .then((sessionCookie) => {
+          currentSessionCookie = sessionCookie;
+          return new Promise((resolve) => setTimeout(() => resolve(
+            admin.auth().revokeRefreshTokens(uid),
+          ), 1000));
+        })
+        .then(() => {
+          return admin.auth().verifySessionCookie(currentSessionCookie)
+            .should.eventually.be.fulfilled;
+        })
+        .then(() => {
+          return admin.auth().verifySessionCookie(currentSessionCookie, true)
+            .should.eventually.be.rejected.and.have.property('code', 'auth/session-cookie-revoked');
+        });
+    }).timeout(5000);
+
+    it('fails when called with an invalid ID token', () => {
```

Comment:
It's probably enough to have tests like these that do not make remote API calls as unit tests.